git-extra/PKGBUILD

```diff
@@ -1,11 +1,15 @@
 # Maintainer: nalla <nalla@hamal.uberspace.de>
 
-pkgname=('git-extra')
+_realname="git-extra"
+pkgbase="mingw-w64-${_realname}"
+pkgname=($_realname
+        "${MINGW_PACKAGE_PREFIX}-${_realname}")
 _ver_base=1.1
-pkgver=1.1.614.d551cf0cc
+pkgver=1.1.616.0ce99c8d1
 pkgrel=1
 pkgdesc="Git for Windows extra files"
-arch=('i686' 'x86_64')
+arch=('any')
```

Comment:
I like the idea of publishing the packages concurrently.

However, we _will_ need to publish the `i686` and `x86_64` variants of `git-extra`, not the `any` variant, because Git for Windows' tooling is unable to handle any package without `mingw-w64-` prefix _and_ without the CPU architecture in its suffix.

I also wondered whether there is a better way wherein the `git-extra` package would be automatically uninstalled and the `mingw-w64-git-extra` package would be installed. On a first read, https://wiki.archlinux.org/title/PKGBUILD#replaces would be the correct solution there. But I fear that won't work for us because we will provide `mingw-w64-git-extra` both in the `git-for-windows` _and_ in the `git-for-windows-mingw32` repository. And if there are _two_ packages that replace the same package, and those two packages conflict with each other, I wonder whether Pacman does the right thing.

But then, we could exclude that package specifically from `git-for-windows-mingw32` (and from `git-for-windows-mingw64`) so that only one or the other is available, depending whether you run Pacman in `git-sdk-32` or `git-sdk-64`.

Having said all that, I think you're right and we can only do that once we have taken care of all of our tooling, for which we need a transitional period where we build both `git-extra` and `mingw-w64-git-extra`.

Back to the `arch` problem: we cannot specify `arch` conditional on the `pkgname`, sadly, as it is a global.

So I fear that we will have to either put in a hack, or copy the `PKGBUILD` file whole-sale. The hack would look somewhat like this:

```sh
if test -z "$BUILD_MINGW_W64_GIT_EXTRA"
then
  pkgname=$_realname
  arch=('i686' 'x86_64')
else
  pkgname=$MINGW_PACKAGE_PREFIX-$_realname
  arch=('any')
fi
```

Reply:
Good catch! Fixed by simply overriding the `arch` in `package_git-extra()` 🎉 I'm really impressed by the flexibility we have in those `PKGBUILD` files.

The resulting package names look good:
- `git-extra-1.1.616.3f26b1b0c-1-x86_64.pkg.tar.zst`
- `mingw-w64-x86_64-git-extra-1.1.616.3f26b1b0c-1-any.pkg.tar.zst`